modin/modin_logging/logger_metaclass.py

```diff
@@ -0,0 +1,133 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains ``LoggerMetaClass`` class.
+
+``LoggerMetaClass`` is used for adding logging to Modin classes.
+"""
+
+from functools import wraps
+from types import FunctionType, MethodType
+
+from .config import get_logger
+from modin.config import LogMode
+
+
+def logger_class_wrapper(classname, name, method):
+    """
+    Execute Modin functions with logging if enabled.
+
+    Parameters
+    ----------
+    classname : str
+        The name of the class the LoggerMetaClass is being applied to.
+    name : str
```

Comment:
```suggestion
    method_name : str
```

Reply:
Are you suggesting to change the function signature as well and not just the doc string?

Reply:
Yep.